### PR TITLE
Fixed issue with processcleanup.sh tool (Axis2 server error)

### PR DESCRIPTION
**Pull request by isurusuranga**
fixed https://wso2.org/jira/browse/BPS-787

### CODE REVIEW

File: modules/samples/product/process-tools/src/main/java/org/wso2/bps/samples/processcleanup/CleanupConstants.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	public static final String DRIVER_CLASS_NAME = "driverClassName";
+	public static final String USER_NAME = "username";
+	public static final String PASSWORD = "password";
+	public static final String FILTER_STATES = "process.filterStates";
+	public static final String CLIENT_TRUST_STORE_PATH = "clientTrustStorePath";
+	public static final String CLIENT_TRUST_STORE_PASSWORD = "clientTrustStorePassword";
+	public static final String CLIENT_TRUST_STORE_TYPE = "clientTrustStoreType";
+	public static final String REG_PATH = "/_system/config/bpel/packages/";
+	public static final String DELETE_INSTANCES = "delete.instances";
+    public static final String TRUE = "true";
```
Comment (nandika) [inline modules/samples/product/process-tools/src/main/java/org/wso2/bps/samples/processcleanup/CleanupConstants.java]: Fix the formatting

File: modules/samples/product/process-tools/src/main/java/org/wso2/bps/samples/processcleanup/CleanupExecutor.java [hunk 2]
```diff
@@ -27,346 +27,417 @@
...
+				                           .getTextContent();
+				databasePassword = document.getDocumentElement()
+				                           .getElementsByTagName(CleanupConstants.PASSWORD).item(i)
+				                           .getTextContent();
+				dbConfigFound = true;
+				break;
+			}
+		}
+		if (!dbConfigFound) {
+			log.info("ERROR: DB configurations not found or invalid!");
```
Comment (vinok88) [inline modules/samples/product/process-tools/src/main/java/org/wso2/bps/samples/processcleanup/CleanupExecutor.java]: this should be an error log.

File: modules/samples/product/process-tools/src/main/java/org/wso2/bps/samples/processcleanup/CleanupConstants.java (hunk 1 above)
Comment (isurusuranga) [inline modules/samples/product/process-tools/src/main/java/org/wso2/bps/samples/processcleanup/CleanupConstants.java]: fixed it

File: modules/samples/product/process-tools/src/main/java/org/wso2/bps/samples/processcleanup/DBQuery.java [hunk 3]
```diff
@@ -21,187 +21,207 @@
...
+			//todo: need to update the queries for H2 DB type
+			System.out.println("H2 not yet supported.");
+			System.exit(0);
+
+		} else {
+			System.out.println("Unsupported DB Type \n" + "or Invalid Driver Name!");
+		}
+	}
+
+	//formats the String with the process name and returns the delete query
```
Comment (vinok88) [inline modules/samples/product/process-tools/src/main/java/org/wso2/bps/samples/processcleanup/DBQuery.java]: Use block comments.

File: modules/samples/product/process-tools/src/main/java/org/wso2/bps/samples/processcleanup/CleanupExecutor.java (hunk 2 above)
Comment (isurusuranga) [inline modules/samples/product/process-tools/src/main/java/org/wso2/bps/samples/processcleanup/CleanupExecutor.java]: fixed in https://github.com/wso2/product-bps/pull/90

File: modules/samples/product/process-tools/src/main/java/org/wso2/bps/samples/processcleanup/DBQuery.java (hunk 3 above)
Comment (isurusuranga) [inline modules/samples/product/process-tools/src/main/java/org/wso2/bps/samples/processcleanup/DBQuery.java]: fixed in https://github.com/wso2/product-bps/pull/90

File: modules/samples/product/process-tools/src/main/java/org/wso2/bps/samples/processcleanup/RegistryCleaner.java [hunk 4]
```diff
@@ -27,71 +27,122 @@
...
+				return true;
+			}
+			/* if the number of deployment units of the given package equals to one, then removes
+			   it from /_system/config/bpel/packages/<package> */
+			else if (count == 1) {
+				resourceAdminServiceStub.delete(regPathAppend);
+				return true;
+			}
+		} catch (Exception e) {
+			e.printStackTrace();
```
Comment (nandika) [inline modules/samples/product/process-tools/src/main/java/org/wso2/bps/samples/processcleanup/RegistryCleaner.java]: Lets remove all printStackTrace() and replace with logs

File: modules/samples/product/process-tools/src/main/java/org/wso2/bps/samples/processcleanup/DBQuery.java [hunk 5]
```diff
@@ -21,187 +21,207 @@
...
+	DBQuery(String databaseURL, String bpsHome) {
+		Properties prop = new Properties();
+
+		try {
+			String configPath =
+					bpsHome + CleanupConstants.REPOSITORY + File.separator + CleanupConstants.CONF +
+					File.separator + CleanupConstants.CLEANUP_PROPERTIES;
+			prop.load(new FileInputStream(configPath));
+		} catch (IOException e) {
+			e.printStackTrace();
```
Comment (nandika) [inline modules/samples/product/process-tools/src/main/java/org/wso2/bps/samples/processcleanup/DBQuery.java]: replace printStackTrace with log

File: modules/samples/product/process-tools/src/main/java/org/wso2/bps/samples/processcleanup/RegistryCleaner.java (hunk 4 above)
Comment (isurusuranga) [inline modules/samples/product/process-tools/src/main/java/org/wso2/bps/samples/processcleanup/RegistryCleaner.java]: Fixed in https://github.com/wso2/product-bps/pull/100

File: modules/samples/product/process-tools/src/main/java/org/wso2/bps/samples/processcleanup/DBQuery.java (hunk 5 above)
Comment (isurusuranga) [inline modules/samples/product/process-tools/src/main/java/org/wso2/bps/samples/processcleanup/DBQuery.java]: Fixed in https://github.com/wso2/product-bps/pull/100

File: modules/samples/product/process-tools/src/main/java/org/wso2/bps/samples/processcleanup/CleanupExecutor.java [hunk 6]
```diff
@@ -27,346 +27,417 @@
 import java.io.*;
 import java.sql.*;
 import java.util.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 
-// The main class
+/**
+ * The main class
```
Comment (Kishanthan) [inline modules/samples/product/process-tools/src/main/java/org/wso2/bps/samples/processcleanup/CleanupExecutor.java]: Have some meaningful javadocs comments. "The main class" does not give an idea on what this class is about and usage.

File: modules/samples/product/process-tools/src/main/java/org/wso2/bps/samples/processcleanup/RegistryCleaner.java [hunk 7]
```diff
@@ -27,71 +27,122 @@
...
+		options.setManageSession(true);
+
+		String userName = prop.getProperty(CleanupConstants.BPS_USER_NAME);
+		String password = prop.getProperty(CleanupConstants.BPS_PASSWORD);
+		String hostName = NetworkUtils.getLocalHostname();
+
+		authenticationAdminStub.login(userName, password, hostName);
+
+		ServiceContext serviceContext = authenticationAdminStub.
+				                                                       _getServiceClient()
```
Comment (Kishanthan) [inline modules/samples/product/process-tools/src/main/java/org/wso2/bps/samples/processcleanup/RegistryCleaner.java]: Is this line formatted properly ?

File: modules/samples/product/process-tools/src/main/java/org/wso2/bps/samples/processcleanup/CleanupExecutor.java (hunk 6 above)
Comment (isurusuranga) [inline modules/samples/product/process-tools/src/main/java/org/wso2/bps/samples/processcleanup/CleanupExecutor.java]: Already fixed. See https://github.com/wso2/product-bps/pull/100

File: modules/samples/product/process-tools/src/main/java/org/wso2/bps/samples/processcleanup/RegistryCleaner.java (hunk 7 above)
Comment (isurusuranga) [inline modules/samples/product/process-tools/src/main/java/org/wso2/bps/samples/processcleanup/RegistryCleaner.java]: Fixed in https://github.com/wso2/product-bps/pull/100

**Pull request merged**